admin/server/projects.go

```diff
@@ -420,6 +414,44 @@ func (s *Server) SetProjectMemberRole(ctx context.Context, req *adminv1.SetProje
 	return &adminv1.SetProjectMemberRoleResponse{}, nil
 }
 
+// fetchInstallationID returns a valid installation ID iff app is installed and user is a collaborator of the repo
+func (s *Server) fetchInstallationID(ctx context.Context, githubURL, userID string) (int64, error) {
+	// Get Github installation ID for the repo
+	installationID, err := s.admin.GetGithubInstallation(ctx, githubURL)
+	if err != nil {
+		if errors.Is(err, admin.ErrGithubInstallationNotFound) {
+			return 0, status.Errorf(codes.PermissionDenied, "you have not granted Rill access to %q", githubURL)
+		}
+
+		return 0, status.Errorf(codes.Internal, "failed to get Github installation: %q", err.Error())
+	}
+
+	if installationID == 0 {
+		return 0, status.Errorf(codes.Internal, "you have not granted Rill access to %q", githubURL)
+	}
+
+	user, err := s.admin.DB.FindUser(ctx, userID)
+	if err != nil {
+		if errors.Is(err, database.ErrNotFound) {
+			return 0, status.Error(codes.PermissionDenied, "does not have permission to create projects")
+		}
```

Comment:
This is kind of an internal error since `claims.OwnerID()` is guaranteed to be a user